crates/bevy_core_pipeline/src/core_2d/main_transparent_pass_2d_node.rs

```diff
@@ -16,20 +16,25 @@ use bevy_utils::tracing::info_span;
 pub struct MainTransparentPass2dNode {}
 
 impl ViewNode for MainTransparentPass2dNode {
-    type ViewQuery = (
-        &'static ExtractedCamera,
-        &'static SortedRenderPhase<Transparent2d>,
-        &'static ViewTarget,
-    );
+    type ViewQuery = (&'static ExtractedCamera, &'static ViewTarget);
 
     fn run<'w>(
         &self,
         graph: &mut RenderGraphContext,
         render_context: &mut RenderContext<'w>,
-        (camera, transparent_phase, target): bevy_ecs::query::QueryItem<'w, Self::ViewQuery>,
+        (camera, target): bevy_ecs::query::QueryItem<'w, Self::ViewQuery>,
         world: &'w World,
     ) -> Result<(), NodeRunError> {
+        let Some(transparent_phases) =
+            world.get_resource::<ViewSortedRenderPhases<Transparent2d>>()
+        else {
+            return Ok(());
+        };
+
         let view_entity = graph.view_entity();
+        let Some(transparent_phase) = transparent_phases.get(&view_entity) else {
+            return Ok(());
```

Comment:
nit: Since this is now a resource and inserted during extract, is this case even possible? It strikes me that even if it can happen, that it would be surprising/erroneous. Might merit a warning and/or `.expect`?

(It looks like most of the new resource accesses follow this pattern.)

Reply:
From what I can tell, Bevy style is to use bailouts like this even when the problem should never happen to reduce unwinding overhead. This may or may not be a good thing, but I try to follow it. We could use `error!` there though.

Reply:
Actually, this will happen normally for the view entities that are created for cascaded shadow maps. Since shadow maps don't render any transparent meshes, their view entities won't have entries in `ViewSortedRenderPhases<Transparent3d>`. This is fine and skipping them is the correct thing to do. So this logic is correct as is, I believe.